woocommerce/woocommerce-gutenberg-products-block
# Enhance: Bring back user to 1st page when filters are changed

**URL page number removal on filter change**: When filters are changed, the page number in the URL is removed. This is to ensure that when a filter is applied, the user is taken back to the first page of results. The code accommodates three different formats of page numbers in the URL and removes them accordingly.

Part of #9876 

### Testing

1. Create a new post.
2. Add all filter blocks and a 'Products Beta' block to it.
3. Save the post and navigate to the frontend.
4. Navigate to the 2nd page using pagination. 
5. Modify any filter and confirm that the page number resets to the 1st page upon filter alteration. 
6. Repeat the above steps and validate that this also holds true for:
    - 'Product Collection' block (The product collection block is currently in the experimental phase and is not shipped with the plugin/core)
    - 'Classic WooCommerce Product Grid Block'
    - Note: This behavior doesn't affect the 'All Products' block.


### WooCommerce Visibility




### Changelog

> Enhancement: Bring back user to 1st page when filters are changed
